### PR TITLE
Add support for `ExtendAccountBatchVersion`

### PR DESCRIPTION
**Pull request by positiveblue**
The `CurrentBatchVersion` has not been updated yet, so this PR contains the changes to make the client compatible with the new batch version but continuous to use the `DefaultBatchVersion`.

### CODE REVIEW

File: auctioneerrpc/auctioneer.proto [hunk 1]
```diff
@@ -792,6 +792,13 @@ message AccountDiff {
     The trader's account key this diff is referring to.
     */
     bytes trader_key = 4;
+
+    /*
+    The new account expiry height used to verify the batch. If the batch is
+    successfully executed the account must update its expiry height to this
+    value.
+    */
+    uint32 new_expiry = 5;
```
Comment (guggero) [inline auctioneerrpc/auctioneer.proto:801]: Reminder for us to tag a new version of the `auctioneerrpc` sub module after this is merged.

File: order/batch.go [hunk 2]
```diff
@@ -47,11 +47,13 @@ const (
...
-	// batch verification protocol. Both server and client should always
-	// refer to this constant. If a client's binary is not updated in time
-	// it will point to a previous version than the server and the mismatch
-	// will be detected during the OrderMatchPrepare call.
+	// ExtendAccountBatchVersion is the first version where accounts expiry
+	// are extended after participating in a batch.
+	ExtendAccountBatchVersion BatchVersion = 1
+
+	// CurrentBatchVersion points to the version used by the client for
+	// verifying a batch. The servers must  always taking this value into
```
Comment (guggero) [inline order/batch.go:55]: nit: weird second sentence.

File: order/batch.go [hunk 3]
```diff
@@ -370,3 +376,9 @@ func DecrementingBatchIDs(startKey, endKey *btcec.PublicKey) []BatchID {
 	}
 	return result
 }
+
+// supportsAccountExtension is a helper function to easily check if a version
+// supports or not account extension after participating in a batch.
+func supportsAccountExtension(version BatchVersion) bool {
```
Comment (guggero) [inline order/batch.go:382]: This could be a method on the `BatchVersion` type itself.

**Review by guggero — COMMENTED** [review]
Looks great at first glance! But will likely need to verify with a server-side PR.

File: auctioneerrpc/auctioneer.proto [hunk 4]
```diff
@@ -1140,6 +1140,12 @@ message TermsResponse {
     the duration buckets currently are in.
     */
     map<uint32, DurationBucketState> lease_duration_buckets = 8;
+
+    /*
+    The value used by the auctioneer to determine if an account expiry height
+    needs to be extended after participating in a batch and for how long.
+    */
+    uint32 auto_renew_extension_blocks = 9;
```
Comment (guggero) [inline auctioneerrpc/auctioneer.proto:1148]: We should probably also add this to the client side `NextBatchInfoResponse` so the user can see the value somewhere.

**Review by guggero — APPROVED** [review]
Great work on this, looking forward to this feature :rocket: 

Just one non-blocking comment, LGTM :tada:

**Review by bhandras — APPROVED** [review]
LGTM, straightfwd changes, great work @positiveblue :+1:

**Pull request merged**